cohttp-eio/src/server.ml

```diff
@@ -114,7 +110,8 @@ let run_domain ssock handler =
   in
   Switch.run (fun sw ->
       while true do
-        Eio.Net.accept_sub ~sw ssock ~on_error (fun ~sw flow _addr ->
+        Eio.Net.accept_fork ~sw ssock ~on_error (fun flow _addr ->
+            Eio.Switch.run @@ fun sw ->
```

Comment:
One last question. What is the reason one needs to nest the two Switch.run? Is it because there is the fork involved and then we want to manage a separate set of fibers?

Reply:
It's because the writer fiber should be tied to the lifetime of this connection, not the lifetime of the server (if the handler function crashes, we want to cancel the writer fiber).

Note that this switch goes away in the next commit anyway (#887).